modules/market/src/main/java/com/opengamma/strata/market/curve/SeasonalNodalCurveDefinition.java

```diff
@@ -0,0 +1,537 @@
+/**
+ * Copyright (C) 2016 - present by OpenGamma Inc. and the OpenGamma group of companies
+ *
+ * Please see distribution for license.
+ */
+package com.opengamma.strata.market.curve;
+
+import java.io.Serializable;
+import java.time.LocalDate;
+import java.time.YearMonth;
+import java.util.Set;
+
+import org.joda.beans.BeanDefinition;
+import org.joda.beans.ImmutableBean;
+import org.joda.beans.ImmutableConstructor;
+import org.joda.beans.Property;
+import org.joda.beans.PropertyDefinition;
+
+import com.google.common.collect.ImmutableList;
+import com.opengamma.strata.basics.ReferenceData;
+import com.opengamma.strata.collect.array.DoubleArray;
+import com.opengamma.strata.market.ShiftType;
+import com.opengamma.strata.market.ValueType;
+import java.util.Map;
+import java.util.NoSuchElementException;
+import org.joda.beans.Bean;
+import org.joda.beans.JodaBeanUtils;
+import org.joda.beans.MetaProperty;
+import org.joda.beans.impl.direct.DirectFieldsBeanBuilder;
+import org.joda.beans.impl.direct.DirectMetaBean;
+import org.joda.beans.impl.direct.DirectMetaProperty;
+import org.joda.beans.impl.direct.DirectMetaPropertyMap;
+
+
+@BeanDefinition
+public final class SeasonalNodalCurveDefinition
```

Comment:
Could this be package scoped since it should not be created directly by end-users